src/lib/utils/lib.ts

```diff
@@ -15,17 +20,39 @@ export const throwIfNotAddress = (address: string) => {
  * @param l1Address
  * @returns
  */
-export const applyL1ToL2Alias = (l1Address: string): BigNumber => {
+export const applyL1ToL2Alias = (l1Address: string): string => {
   throwIfNotAddress(l1Address)
-  return BigNumber.from(l1Address).add(ADDRESS_ALIAS_OFFSET)
+
+  // we use BigInts in here and undo to allow for proper under/overflow behaviour
+  // BigInt.asUintN calculates the correct positive modulus
+  return getAddress(
+    '0x' +
+      BigInt.asUintN(
+        ADDRESS_BIT_LENGTH,
+        BigInt(l1Address) + ADDRESS_ALIAS_OFFSET_BIG_INT
+      )
+        .toString(16)
+        .padStart(ADDRESS_NIBBLE_LENGTH, '0')
+  )
 }
 
 /**
  * Find the L1 alias of an L2 address
  * @param l2Address
  * @returns
  */
-export const undoL1ToL2Alias = (l2Address: string): BigNumber => {
+export const undoL1ToL2Alias = (l2Address: string): string => {
   throwIfNotAddress(l2Address)
-  return BigNumber.from(l2Address).sub(ADDRESS_ALIAS_OFFSET)
+
+  // we use BigInts in here and apply to allow for proper under/overflow behaviour
+  // BigInt.asUintN calculates the correct positive modulus
+  return getAddress(
+    '0x' +
+      BigInt.asUintN(
+        ADDRESS_BIT_LENGTH,
+        BigInt(l2Address) - ADDRESS_ALIAS_OFFSET_BIG_INT
+      )
+        .toString(16)
+        .padStart(ADDRESS_NIBBLE_LENGTH, '0')
```

Comment:
has a bit of repeated code between here and the apply function.
the only difference with them is either the `-` or `+`

could have an intermediate function that is used with both

Reply:
updated